gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -423,16 +422,14 @@ def self.run_all_sig_blocks
   # the module target is adding the methods from the module source to itself. we need to check that for all instance
   # methods M on source, M is not defined on any of target's ancestors.
   def self._hook_impl(target, singleton_class, source)
-    target_was_final = module_with_final?(target)
-    if !target_was_final && !module_with_final?(source)
-      return
-    end
     # we do not need to call add_was_ever_final here, because we have already marked
     # any such methods when source was originally defined.
-    add_module_with_final(target)
-    install_hooks(target)
-
-    if !target_was_final
+    if !module_with_final?(target)
+      if !module_with_final?(source)
```

Comment:
Is it worth inlining the use of the hash here to avoid pushing function frames?
```suggestion
    if !@modules_with_final.has_key?(target)
      if !@modules_with_final.has_key?(source)
```

Reply:
Did this for all uses of `module_with_final?` and then removed the function, for consistency.